src/objective-c/GRPCClient/GRPCCall+OAuth2.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ * Copyright 2015, Google Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#import "GRPCCall.h"
+
+// Helpers for setting and reading headers compatible with OAuth2.
+@interface GRPCCall (OAuth2)
+
+// Setting this property is equivalent to setting "Bearer <passed token>" as the value of the
+// request header with key "authorization" (the authorization header). Setting it to nil removes the
+// authorization header from the request.
+// The value obtained by getting the property is the OAuth2 bearer token if the authorization header
+// of the request has the form "Bearer <token>", or nil otherwise.
+@property(atomic, copy) NSString *oauth2_accessToken;
```

Comment:
LGTM

On Fri, Aug 7, 2015 at 10:15 AM, Jorge Canizales <notifications@github.com>
wrote:

> In src/objective-c/GRPCClient/GRPCCall+OAuth2.h
> <https://github.com/grpc/grpc/pull/2838#discussion_r36539684>:
>
> > + * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
> > + * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> > + *
> > + */
> > +
> > +#import "GRPCCall.h"
> > +
> > +// Helpers for setting and reading headers compatible with OAuth2.
> > +@interface GRPCCall (OAuth2)
> > +
> > +// Setting this property is equivalent to setting "Bearer <passed token>" as the value of the
> > +// request header with key "authorization" (the authorization header). Setting it to nil removes the
> > +// authorization header from the request.
> > +// The value obtained by getting the property is the OAuth2 bearer token if the authorization header
> > +// of the request has the form "Bearer <token>", or nil otherwise.
> > +@property(atomic, copy) NSString *oauth2_accessToken;
>
> From a quick search of ObjC methods on the wild that start with "oauth",
> I've gone with oauth2AccessToken (721b7a3
> <https://github.com/grpc/grpc/commit/721b7a3923996c7ac8aa7258f1fe7d5bc416af2b>
> ).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/2838/files#r36539684>.
>